protocol/src/test/kotlin/OfferingTest.kt

```diff
@@ -0,0 +1,51 @@
+package protocol
+
+import assertk.assertAll
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import models.CurrencyDetails
+import models.Offering
+import models.OfferingData
+import models.PresentationExchange
+import kotlin.test.Test
+
+class OfferingTest {
+  @Test
+  fun `can create a new offering`() {
+    val offering = Offering.create(
+      from = "from",
+      OfferingData(
+        description = "my fake offering",
+        payoutUnitsPerPayinUnit = 1,
+        payinCurrency = CurrencyDetails("", "", ""),
+        payoutCurrency = CurrencyDetails("", "", ""),
+        payinMethods = listOf(),
+        payoutMethods = listOf(),
+        requiredClaims = PresentationExchange()
+      )
+    )
+
+    assertAll {
+      assertThat(offering.data.description).isEqualTo("my fake offering")
+      assertThat(offering.metadata.id.prefix).isEqualTo("offering")
+    }
+  }
+
+  @Test
+  fun `sign populates offering signature`() {
+    val offering = TestData.getOffering()
+    offering.sign("fakepk", "fakekid")
+
+    assertThat(offering.signature).isEqualTo("blah")
+  }
+
+  @Test
+  fun `can parse offering from a json string`() {
+    val offering = TestData.getOffering()
+    offering.sign("fakepk", "fakekid")
+    val jsonResource = offering.toString()
```

Comment:
Technically, your "parse" test is also testing the stringification functionality

It's slightly more correct to just hard code a JSON string here in the test

But, it's splitting hairs a bit and we have bigger fish to fry so I like the pragmatism